core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -163,14 +168,21 @@ public KafkaSchemaRegistry(SchemaRegistryConfig config,
    * of say the last, is arbitrary.
    */
   // TODO: once RestConfig.PORT_CONFIG is deprecated, remove the port parameter.
-  static int getPortForIdentity(int port, List<String> configuredListeners) {
+  static AbstractMap.SimpleEntry<String, Integer> getSchemeAndPortForIdentity(
+      int port,
+      List<String> configuredListeners) {
     List<URI> listeners = Application.parseListeners(configuredListeners, port,
                                                      Arrays.asList("http", "https"), "http");
-    return listeners.get(0).getPort();
+    for (URI listener: listeners) {
```

Comment:
This just always prefers https if it is availalble. shouldn't we only do that if we have at least a truststore? or is the intent that if you have https available you *must* use it in order to avoid configuration errors?

i'm thinking about the upgrade path and wondering if always using https will make it impossible to do a zero downtime upgrade to https

Reply:
The one option I can think of is using a config similar to kafka's inter broker protocol.  If that's not available we can may be fallback to default http? As far as Truststore, I guess it uses default JVM cacerts as the truststore.

Reply:
as discussed offline, this makes sense to me.